how-to/customize-home-templates/client/src/integrations/emoji/integration-provider.ts

```diff
@@ -0,0 +1,192 @@
+import * as emoji from "node-emoji";
+
+/**
+ * Implement the integration provider for Emojis.
+ */
+export class EmojiIntegrationProvider implements IntegrationModule<EmojiSettings> {
+    /**
+     * Provider id.
+     * @internal
+     */
+    private static readonly _PROVIDER_ID = "emoji";
+
+    /**
+     * The key to use for a emoji result.
+     * @internal
+     */
+    private static readonly _EMOJI_PROVIDER_DETAILS_ACTION = "Emoji Details";
+
+    /**
+     * The key to use for a emoji copy key action.
+     * @internal
+     */
+    private static readonly _EMOJI_PROVIDER_COPY_KEY_ACTION = "Copy Key";
+
+    /**
+     * The key to use for a emoji copy key action.
+     * @internal
+     */
+    private static readonly _EMOJI_PROVIDER_COPY_EMOJI_ACTION = "Copy Emoji";
+
+    /**
+    * The integration manager.
+    * @internal
+    */
+    private _integrationManager: IntegrationManager | undefined;
+
+    /**
+     * The module is being registered.
+     * @param integrationManager The manager for the integration.
+     * @param integration The integration details.
+     * @returns Nothing.
+     */
+    public async register(
+        integrationManager: IntegrationManager,
+        integration: Integration<EmojiSettings>
+    ): Promise<void> {
+        this._integrationManager = integrationManager;
+    }
+
+    /**
+     * The module is being deregistered.
+     * @param integration The integration details.
+     * @returns Nothing.
+     */
+    public async deregister(integration: Integration<EmojiSettings>): Promise<void> {
+    }
+
+    /**
+     * Get a list of the static application entries.
+     * @param integration The integration details.
+     * @returns The list of application entries.
+     */
+    public async getAppSearchEntries(integration: Integration<EmojiSettings>): Promise<HomeSearchResult[]> {
+        const results = [];
+
+        return results;
+    }
+
+    /**
+     * An entry has been selected.
+     * @param integration The integration details.
+     * @param result The dispatched result.
+     * @param lastResponse The last response.
+     * @returns True if the item was handled.
+     */
+    public async itemSelection(
+        integration: Integration<EmojiSettings>,
+        result: CLIDispatchedSearchResult,
+        lastResponse: CLISearchListenerResponse
+    ): Promise<boolean> {
+        if (result.action.name === EmojiIntegrationProvider._EMOJI_PROVIDER_COPY_EMOJI_ACTION && result.data.emoji) {
+            await fin.Clipboard.writeText({ data: result.data.emoji });
+            return true;
+        } else if (result.action.name === EmojiIntegrationProvider._EMOJI_PROVIDER_COPY_KEY_ACTION && result.data.key) {
+            await fin.Clipboard.writeText({ data: result.data.key });
+            return true;
+        } else if (result.action.name === EmojiIntegrationProvider._EMOJI_PROVIDER_DETAILS_ACTION && result.data.url) {
+            await fin.System.openUrlWithBrowser(result.data.url);
```

Comment:
shouldn't this use the injected launch url function?

Reply:
await this._integrationManager.openUrl(result.data.url);

Reply:
yup it should, missed that one